### PR TITLE
fix: (menu & menuitem): vertical menu theming updates

### PR DESCRIPTION
**Pull request by jaanus03**
The aim of this this PR is to update vertical menu styles to make it look great again.

**Current vertical menu styles:** 
screenshot 2019-02-20 at 14 30 13
screenshot 2019-02-20 at 14 38 26
screenshot 2019-02-20 at 14 38 42

**New vertical menu styles:**
screenshot 2019-02-20 at 14 43 16
screenshot 2019-02-20 at 14 43 04
screenshot 2019-02-20 at 14 42 46

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,6 +44,32 @@
       "git add"
     ]
   },
+  "repository": {
```
Comment (bmdalex) [inline package.json]: wrong merge, please take a look

File: packages/react/src/themes/teams-dark/siteVariables.ts [hunk 2]
```diff
@@ -33,7 +33,10 @@ export const green04 = naturalColors.lightGreen[900]
 //
 // SHADOW LEVELS
 //
-export const shadowLevel1 = '0 .2rem .4rem -.075rem rgba(255, 255, 255, 0.1)'
+export const shadowLevel1 = '0 .2rem .4rem -.075rem rgba(0, 0, 0, .1)'
```
Comment (bmdalex) [inline packages/react/src/themes/teams-dark/siteVariables.ts]: where are these values coming from? can you reference a design file / spec in a private chat?

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts [hunk 3]
```diff
@@ -1,13 +1,26 @@
+import { pxToRem } from '../../../../lib'
+
 export interface MenuVariables {
   color: string
   activeColor: string
   activeBackgroundColor: string
+  selectedBackgroundColor: string
```
Comment (bmdalex) [inline packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: what's the difference between `selectedBackgroundColor` and `activeBackgroundColor` we should always aim for meaningful names

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
 export default (siteVars: any): MenuVariables => {
   return {
     color: siteVars.white,
     activeColor: siteVars.black,
     activeBackgroundColor: siteVars.accessibleYellow,
+    selectedBackgroundColor: siteVars.accessibleCyan,
+
+    verticalMenuBackgroundColor: siteVars.colors.black,
+
+    menuDividerHeight: pxToRem(2),
```
Comment (bmdalex) [inline packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: NIT: ~~`menuDividerHeight`~~ -> `dividerHeight` since we're already in the context of the menu

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts [hunk 5]
```diff
@@ -1,13 +1,26 @@
...
+    menuBorderWidth: pxToRem(2),
```
Comment (bmdalex) [inline packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: NIT: ~~`menuBorderWidth`~~ -> `borderWidth` since we're already in the context of the menu

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts [hunk 6]
```diff
@@ -1,13 +1,26 @@
+import { pxToRem } from '../../../../lib'
+
 export interface MenuVariables {
   color: string
   activeColor: string
   activeBackgroundColor: string
+  selectedBackgroundColor: string
+
+  verticalMenuBackgroundColor: string
+  menuDividerHeight: string
```
Comment (bmdalex) [inline packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: NIT: ~~`menuDividerHeight`~~ -> `dividerHeight` since we're already in the context of the menu

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts [hunk 7]
```diff
@@ -1,13 +1,26 @@
+import { pxToRem } from '../../../../lib'
+
 export interface MenuVariables {
   color: string
   activeColor: string
   activeBackgroundColor: string
+  selectedBackgroundColor: string
+
+  verticalMenuBackgroundColor: string
```
Comment (bmdalex) [inline packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: NIT: ~~`verticalMenuBackgroundColor `~~ -> `verticalBackgroundColor ` since we're already in the context of the menu

File: packages/react/src/themes/teams/components/Menu/menuVariables.ts [hunk 8]
```diff
@@ -21,38 +25,51 @@ export interface MenuVariables {
   primaryUnderlinedBorderColor: string
 
   disabledColor: string
-
   lineHeightBase: string
-
   horizontalPadding: string
+
+  verticalMenuBackgroundColor: string
```
Comment (bmdalex) [inline packages/react/src/themes/teams/components/Menu/menuVariables.ts]: NIT: ~~`verticalMenuBackgroundColor `~~ -> `verticalBackgroundColor ` since we're already in the context of the menu

same for `verticalMenuItemPadding`, `verticalMenuBoxShadow`, `menuDividerHeight `, `menuBorderWidth `

File: yarn.lock [hunk 9]
```diff
@@ -138,6 +138,13 @@
     "@babel/helper-plugin-utils" "^7.0.0"
     "@babel/plugin-transform-typescript" "^7.1.0"
 
+"@babel/runtime@^7.0.0-beta.49":
```
Comment (bmdalex) [inline yarn.lock]: looks like wrong merge, pls revert

**Review by bmdalex — DISMISSED** [review]
nice changes, some comments to address

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts (hunk 3 above)
Comment (jaanus03) [reply packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: selectedBackgroundColor: selected/preselected item in the menu
activeBackgroundColor: hover & focus state of the menuitem

Happy to change the names to something else if you've a recommendation.

File: packages/react/src/themes/teams-dark/siteVariables.ts (hunk 2 above)
Comment (jaanus03) [reply packages/react/src/themes/teams-dark/siteVariables.ts]: Pinged in private.

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts (hunk 3 above)
Comment (bmdalex) [reply packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: got it

> selectedBackgroundColor: selected/preselected item in the menu

We're using the `active` prop to describe selected items in the menu, so a better name would actually be `activeBackgroundColor`

Now that name is used for hover & focus state already and it was there before this PR was introduced so it was an error on our side.

Therefore, I'd recommend changing:
`activeBackgroundColor` -> `focusedBackgroundColor` (which already exists, can we reuse it) if not we need to understand why
`selectedBackgroundColor` -> `activeBackgroundColor`

What do you think @jaanus03 ?

Comment (jaanus03) [reply packages/react/src/themes/teams-high-contrast/components/Menu/menuVariables.ts]: Thanks @Bugaa92, this makes perfect sense. I've updated the names.

**Comment by bmdalex** [comment]
@jaanus03 

I found some time to take the branch and test it. It's true that the Dark and HC themes are much better, but we have a lot of regression on the general theme..

### 1. wrong fonts for regular theme (`master` vs your branch):

screenshot 2019-02-22 at 13 19 24


### 2. `underlined` menus styles broken:

screen recording 2019-02-22 at 13 23 27

### 3. `primary underlined` menus styles COMPLETELY broken:

screen recording 2019-02-22 at 13 35 11

### 4. `iconOnly` menus styles broken:

screen recording 2019-02-22 at 13 25 37

### 5. `vertical pointing` menus focused state broken:

screen recording 2019-02-22 at 13 31 43

### 6. focus outline broken for horizontal (no `vertical` prop) menus:

screen recording 2019-02-22 at 12 52 27

### 7. even in HC this TabList menu should have other colors I think:

screen recording 2019-02-22 at 13 13 41


@codepretty could you help validating some of these, I know you we're keeping a close eye on menus

**Comment by layershifter** [comment]
I think that issue with fonts comes from #950 and will be fixed by #951.

File: packages/react/src/themes/teams-dark/siteVariables.ts (hunk 2 above)
Comment (codepretty) [inline packages/react/src/themes/teams-dark/siteVariables.ts]: If these values are the same as teams theme, you shouldn't need to specify here

File: packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts [hunk 10]
```diff
@@ -1,9 +1,19 @@
...
+  focusedBackgroundColor: 'transparent',
+
+  hoverBackgroundColor: siteVars.gray08,
+
+  activeColor: siteVars.colors.white,
+  activeBackgroundColor: siteVars.gray08,
+
+  verticalBackgroundColor: siteVars.gray10,
+
+  dividerHeight: pxToRem(2),
```
Comment (codepretty) [inline packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts]: Should divider height and border width really different between default and dark themes? Usually, it's just color differences.

**Comment by codepretty** [comment]
Looks like the menu of the doc site was changed
image

@levithomason do you know how @jaanus03 can exclude his changes from affecting the left rail menu?

--- 
Update: chatted with @levithomason about this and he says it is fine to check in

File: packages/react/src/themes/teams/components/Menu/menuStyles.ts [hunk 11]
```diff
@@ -10,18 +10,26 @@ import {
...
     return {
       display: 'flex',
+      minHeight: pxToRem(24),
+      margin: 0,
+      padding: 0,
+      listStyleType: 'none',
       ...(iconOnly && { alignItems: 'center' }),
       ...(vertical && {
         flexDirection: 'column',
+        boxShadow: variables.verticalBoxShadow,
```
Comment (codepretty) [inline packages/react/src/themes/teams/components/Menu/menuStyles.ts]: I don't think we want box shadow for all vertical menus, just submenus

**Comment by codepretty** [comment]
Hover and keyboard focus background colors are supposed to be different
image

File: packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts (hunk 10 above)
Comment (jaanus03) [reply packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts]: That's what I got from the redlines. Correct me if I'm wrong.

**Comment by codepretty** [comment]
> Looks like the menu of the doc site was changed
> image

@levithomason do you know how @jaanus03 can exclude his changes from affecting the left rail menu?

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 12]
```diff
@@ -62,8 +62,15 @@ const getFocusedStyles = ({
         }
       : {
           color,
-          background: v.focusedBackgroundColor,
+          background: v.hoverBackgroundColor,
         }),
+
+    ...(vertical && isFromKeyboard && !pointing && !primary
```
Comment (codepretty) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts:68]: could this just work for all non iconOnly and non pointing menus, meaning the default horizontal menu? it's not worth a lot of work to get it working because we don't have that scenario in Teams to my understanding, but if it just works or is close that would be nice.

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 13]
```diff
@@ -74,6 +81,7 @@ const itemSeparator: ComponentSlotStyleFunction<MenuItemPropsAndState, MenuVaria
   const { iconOnly, pointing, pills, primary, underlined, vertical } = props
 
   return (
+    !vertical &&
     !pills &&
     !underlined &&
     !(pointing && vertical) &&
```
Comment (codepretty) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: do you still need the !vertical in line 87 since you added !vertical in 84?

**Comment by codepretty** [comment]
The focus keyboard state should have a double border for consistent accessibility, 1px white (inner) && 1px black (outer), for both default and dark themes

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts (hunk 12 above)
Comment (jaanus03) [reply packages/react/src/themes/teams/components/Menu/menuItemStyles.ts:68]: To be honest it doesn't look that great on other use cases because of the borders/underlines there already.

It's worth coming back to this when we've an understanding where we go with the menu variations altogether. The component styles should be rewritten from ground up once we've mapped our needs.

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts (hunk 13 above)
Comment (jaanus03) [reply packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: Good catch. No we don't.

**Comment by jaanus03** [comment]
> The focus keyboard state should have a double border for consistent accessibility, 1px white (inner) && 1px black (outer), for both default and dark themes

Done.

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 14]
```diff
@@ -62,28 +62,38 @@ const getFocusedStyles = ({
...
           color,
-          background: v.focusedBackgroundColor,
+          background: v.hoverBackgroundColor,
         }),
+
+    ...(vertical && isFromKeyboard && !pointing && !primary
+      ? {
+          border: v.focusedBorder,
+          outline: v.focusedOutline,
+          margin: '1px',
```
Comment (codepretty) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: should this be `pxToRem(1)`?

File: docs/src/examples/components/Menu/Usages/MenuExampleToolbarWithSubmenu.shorthand.tsx [hunk 15]
```diff
@@ -0,0 +1,184 @@
+import * as React from 'react'
```
Comment (codepretty) [inline docs/src/examples/components/Menu/Usages/MenuExampleToolbarWithSubmenu.shorthand.tsx]: In retrospect, I think just adding a more complex menu to the original example of **Menu as a Toolbar** would suffice. I don't think we need a new usage example for **Menu as a Toolbar with submenu**.

**Review by codepretty — APPROVED** [review]

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts (hunk 14 above)
Comment (jaanus03) [reply packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: Yup. Fixed it.

File: docs/src/examples/components/Menu/Usages/MenuExampleToolbarWithSubmenu.shorthand.tsx (hunk 15 above)
Comment (jaanus03) [reply docs/src/examples/components/Menu/Usages/MenuExampleToolbarWithSubmenu.shorthand.tsx]: Done.

File: docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx [hunk 16]
```diff
@@ -76,15 +76,91 @@ const items = [
     indicator: false,
     menu: {
       items: [
-        { key: '1', content: 'item1' },
+        {
+          key: '1',
+          content: 'item1',
+          icon: {
+            name: 'bookmark',
+            variables: { outline: true },
```
Comment (layershifter) [inline docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx]: Can be just `outline: true` (#1002):
```suggestion
            outline: true,
```

File: packages/react/src/lib/felaRenderer.tsx [hunk 17]
```diff
@@ -16,6 +16,7 @@ const filterClassName = (className: string): boolean =>
 
 const createRendererConfig = (options: any = {}) => ({
   devMode: process.env.NODE_ENV !== 'production',
+  selectorPrefix: '_',
```
Comment (layershifter) [inline packages/react/src/lib/felaRenderer.tsx]: Wrong merge, please remove it

File: docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx [hunk 18]
```diff
@@ -76,15 +76,91 @@ const items = [
...
+          key: 'divider2',
+          kind: 'divider',
+        },
+        {
+          key: '8',
+          content: 'item3',
+          icon: {
+            name: 'trash-can',
+            outline: true,
+          },
```
Comment (layershifter) [inline docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx:117]: image


These items looks quite strange, should we remove one of them?

File: CHANGELOG.md [hunk 19]
```diff
@@ -33,6 +33,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Fixed emoji `Icon` spacing issue and added settings `Icon` ([#991](https://github.com/stardust-ui/react/pull/991))
 - Call update `node` if it was changed for `Ref` component @layershifter ([#993](https://github.com/stardust-ui/react/pull/993))
 - Close previous `Popup` on enter key @jongsue ([#985](https://github.com/stardust-ui/react/pull/985))
+- Improve `Menu` styling, vertical menu in particular ([#934](https://github.com/stardust-ui/react/pull/934))
```
Comment (mnajdova) [inline CHANGELOG.md]: Replace with: 
Improve `Menu` styling - vertical menu in Teams theme ([#934](https://github.com/stardust-ui/react/pull/934))

File: packages/react/src/themes/teams-high-contrast/components/Menu/menuItemStyles.ts [hunk 20]
```diff
@@ -1,27 +1,50 @@
...
-import { MenuVariables } from './menuVariables'
+import { MenuVariables } from '../../../teams/components/Menu/menuVariables'
 import { MenuItemProps, MenuItemState } from '../../../../components/Menu/MenuItem'
 
 type MenuItemPropsAndState = MenuItemProps & MenuItemState
 
 const menuItemStyles: ComponentSlotStylesInput<MenuItemPropsAndState, MenuVariables> = {
   wrapper: ({ props, variables }): ICSSInJSStyle => {
-    const { iconOnly, isFromKeyboard } = props
+    const { iconOnly, isFromKeyboard, vertical, active, underlined, primary } = props
```
Comment (layershifter) [inline packages/react/src/themes/teams-high-contrast/components/Menu/menuItemStyles.ts]: We considered to use `p` & `v` pattern.

```diff
-wrapper: ({ props, variables }): ICSSInJSStyle => {
+wrapper: ({ props: p, variables: v }): ICSSInJSStyle => {
```
```diff
-color: variables.activeColor,
+color: v.activeColor,
```
```diff
-...(active && {
+...(p.active && {

File: packages/react/src/themes/teams/components/Menu/menuDividerStyles.ts [hunk 21]
```diff
@@ -8,10 +8,13 @@ const menuDividerStyles: ComponentSlotStylesInput<MenuDividerProps, MenuVariable
     const borderType = p.vertical ? 'borderTop' : 'borderLeft'
 
     return {
-      [borderType]: `1px solid ${borderColor}`,
+      [borderType]: `${v.dividerHeight} solid ${borderColor}`,
       ...(!p.vertical && {
         alignSelf: 'stretch',
       }),
+      ...(p.vertical && {
+        margin: '8px 0',
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Menu/menuDividerStyles.ts]: Oops, please add a variable there and use `pxToRem()` in `menuDividerVariables`

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 22]
```diff
@@ -62,28 +62,38 @@ const getFocusedStyles = ({
...
     !underlined &&
-    !(pointing && vertical) &&
     !iconOnly && {
       '::before': {
         position: 'absolute',
         content: '""',
         top: 0,
         right: 0,
-        ...(vertical ? { width: '100%', height: '1px' } : { width: '1px', height: '100%' }),
+        width: '1px',
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: `pxToRem()`?

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 23]
```diff
@@ -168,6 +178,10 @@ const menuItemStyles: ComponentSlotStylesInput<MenuItemPropsAndState, MenuVariab
         background: 'salmon',
       }),
 
+      ...(vertical && {
+        border: 'solid 2px transparent',
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: Variable?

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 24]
```diff
@@ -401,7 +403,7 @@ const menuItemStyles: ComponentSlotStylesInput<MenuItemPropsAndState, MenuVariab
     }
   },
 
-  menu: ({ props: { vertical } }) => ({
+  menu: ({ props: { vertical }, variables: v }) => ({
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: `p` & `v`, please :)

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 25]
```diff
@@ -274,7 +274,7 @@ const menuItemStyles: ComponentSlotStylesInput<MenuItemPropsAndState, MenuVariab
     }
   },
 
-  root: ({ props, variables: v, theme }): ICSSInJSStyle => {
+  root: ({ props, variables: v }): ICSSInJSStyle => {
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts]: `p` & `v`, please :)

File: packages/react/src/themes/teams/components/Menu/menuStyles.ts [hunk 26]
```diff
@@ -10,18 +10,28 @@ import {
...
-const solidBorder = (color: string) => ({
-  border: `1px solid ${color}`,
-})
-
 export default {
   root: ({ props, variables }): ICSSInJSStyle => {
-    const { iconOnly, fluid, pointing, pills, primary, underlined, vertical } = props
+    const { iconOnly, fluid, pointing, pills, primary, underlined, vertical, submenu } = props
+
+    const solidBorder = (color: string) => ({
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Menu/menuStyles.ts]: I consider to remove `solidBorder()` and inline these styles.

```diff
-...solidBorder(variables.borderColor),
+border: `${v.borderWidth} solid ${v.borderColor}`,
```
```diff
-...solidBorder(variables.primaryBorderColor),
+border: `${v.borderWidth} solid ${v.primaryBorderColor}`,
```

**Comment by layershifter** [comment]
image

We also should remove `borderRadius` in the Sidebar and correct background color.

File: docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx (hunk 18 above)
Comment (jaanus03) [reply docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx:117]: Sure. Removed one.

File: CHANGELOG.md (hunk 19 above)
Comment (jaanus03) [reply CHANGELOG.md]: Done.

**Comment by mnajdova** [comment]
@jaanus03 please update with the latest master, and make sure the screen differences are intentional. Let me know if you don't have access to screener.

File: yarn.lock [hunk 27]
```diff
@@ -13496,4 +13496,4 @@ yauzl@2.4.1:
 yn@^2.0.0:
   version "2.0.0"
   resolved "https://registry.yarnpkg.com/yn/-/yn-2.0.0.tgz#e5adabc8acf408f6385fc76495684c88e6af689a"
-  integrity sha1-5a2ryKz0CPY4X8dklWhMiOavaJo=
+  integrity sha1-5a2ryKz0CPY4X8dklWhMiOavaJo=
```
Comment (mnajdova) [inline yarn.lock]: Please revert the changes in yarn.lock

File: packages/react/src/themes/teams/components/Menu/menuVariables.ts [hunk 28]
```diff
@@ -2,13 +2,19 @@ import { pxToRem } from '../../../../lib'
 
 export interface MenuVariables {
   color: string
+  borderColor: string
+  verticalBorderColor: string
 
-  activeColor: string
-  activeBackgroundColor: string
+  focusedBorder: string
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Menu/menuVariables.ts:8]: These are breaking changes. Please in the description of the PR explain how the usages of the removed variables should be handle.

File: docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx [hunk 29]
```diff
@@ -66,26 +66,60 @@ const items = [
...
+          content: 'item1',
+          icon: {
+            name: 'bookmark',
+            outline: true,
+          },
+        },
+        {
+          key: 'divider',
+          kind: 'divider',
+          isInSubMenu: true,
```
Comment (mnajdova) [inline docs/src/examples/components/Menu/Usages/MenuExampleToolbar.shorthand.tsx]: This should be provided as a prop when the `MenuDivider` is created inside the `Menu` component, so that the users don't have to provide it..

File: CHANGELOG.md [hunk 30]
```diff
@@ -35,6 +35,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Close previous `Popup` on enter key @jongsue ([#985](https://github.com/stardust-ui/react/pull/985))
 - Fixed color of the `timestamp` in `ChatMessages` marked as `mine` in Teams theme @mnajdova ([#1010](https://github.com/stardust-ui/react/pull/1010))
 - Updated table `Icon` & search `Icon` ([#1011](https://github.com/stardust-ui/react/pull/1011))
+- Improve `Menu` styling, vertical menu in Teams theme ([#934](https://github.com/stardust-ui/react/pull/934))
```
Comment (mnajdova) [inline CHANGELOG.md:39]: Add one line in the BREAKING CHANGES as well.

File: packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts [hunk 31]
```diff
@@ -1,9 +1,16 @@
-export interface MenuVariables {
-  color: string
-}
-
-export default (siteVars: any): MenuVariables => {
-  return {
-    color: siteVars.colors.white,
```
Comment (mnajdova) [inline packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts:7]: Can you suggest why the color from here was removed?

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts [hunk 32]
```diff
@@ -253,18 +265,6 @@ const menuItemStyles: ComponentSlotStylesInput<MenuItemPropsAndState, MenuVariab
           }),
       },
 
-      ':last-child': {
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Menu/menuItemStyles.ts:256]: Why was this removed? It's really hard to guess what the changes are as the screener differences are already accepted.. Can you provide some screenshot of why this is not needed anymore?

File: packages/react/src/themes/teams/components/Menu/menuStyles.ts [hunk 33]
```diff
@@ -32,19 +38,15 @@ export default {
...
+          border: `${v.borderWidth} solid ${v.borderColor}`,
           ...(primary && {
-            ...solidBorder(variables.primaryBorderColor),
+            border: `${v.borderWidth} solid ${v.primaryBorderColor}`,
           }),
           borderRadius: pxToRem(4),
         }),
       ...(underlined && {
-        borderBottom: `2px solid ${variables.primaryUnderlinedBorderColor}`,
+        borderBottom: `2px solid ${v.primaryUnderlinedBorderColor}`,
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Menu/menuStyles.ts]: Please always user pxToRem, moreover, this should probably be variable...

File: packages/react/src/themes/teams/components/Menu/menuVariables.ts (hunk 28 above)
Comment (jaanus03) [reply packages/react/src/themes/teams/components/Menu/menuVariables.ts:8]: These were not removed, I rearranged the list of variables to make it better readable.

File: packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts (hunk 31 above)
Comment (jaanus03) [reply packages/react/src/themes/teams-dark/components/Menu/menuVariables.ts:7]: I removed the simplistic interface & imported the real reusable one. It still has the color attribute and all the other related ones too.

File: packages/react/src/themes/teams/components/Menu/menuItemStyles.ts (hunk 32 above)
Comment (jaanus03) [reply packages/react/src/themes/teams/components/Menu/menuItemStyles.ts:256]: This block of styles only applied to vertical menu items which had new set of styles applied.
The vertical menu items don't have border radius any longer, radius is applied to parent "Menu" element instead.

File: packages/react/src/components/Menu/Menu.tsx [hunk 34]
```diff
@@ -166,6 +166,7 @@ class Menu extends AutoControlledComponent<ReactProps<MenuProps>, MenuState> {
             vertical,
             variables,
             styles: styles.divider,
+            inSubmenu: submenu,
```
Comment (mnajdova) [inline packages/react/src/components/Menu/Menu.tsx:169]: 👍

File: packages/react/src/components/Menu/MenuDivider.tsx [hunk 35]
```diff
@@ -30,6 +30,7 @@ export interface MenuDividerProps
   vertical?: boolean
   primary?: boolean
   secondary?: boolean
+  isInSubMenu?: boolean
```
Comment (mnajdova) [inline packages/react/src/components/Menu/MenuDivider.tsx]: We should rename now this to `inSubmenu`

File: packages/react/src/components/Menu/MenuDivider.tsx [hunk 36]
```diff
@@ -52,6 +53,7 @@ class MenuDivider extends UIComponent<ReactProps<MenuDividerProps>> {
     primary: PropTypes.bool,
     secondary: PropTypes.bool,
     vertical: PropTypes.bool,
+    isInSubMenu: PropTypes.bool,
```
Comment (mnajdova) [inline packages/react/src/components/Menu/MenuDivider.tsx]: We should rename now this to `inSubmenu`

Comment (jaanus03) [reply packages/react/src/components/Menu/MenuDivider.tsx]: Done

**Review by mnajdova — APPROVED** [review]
